### PR TITLE
Merge master into branch-2.0

### PR DESCRIPTION
**Pull request by dbczumar**
Signed-off-by: dbczumar <corey.zumar@databricks.com>

## What changes are proposed in this pull request?

Merge master into branch-2.0

## How is this patch tested?

Existing CI

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [X] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/sagemaker/__init__.py [hunk 1]
```diff
@@ -260,19 +259,19 @@ def _deploy(
...
+                               'SecurityGroupIds': [
+                                   'sg-123456abc',
+                               ],
+                               'Subnets': [
+                                   'subnet-123456abc',
+                               ]
+                            }
+                            mfs.deploy(..., vpc_config=vpc_config)
 
     :param data_capture_config: A dictionary specifying the data capture configuration to use when
```
Comment (harupy) [inline mlflow/sagemaker/__init__.py:276]: Can we remove this? We have two `data_capture_config` param descriptions.

Comment (dbczumar) [reply mlflow/sagemaker/__init__.py]: Removed! :)

**Comment by harupy** [comment]
Should we revert the changes in examples/pipelines/sklearn_regression?

File: tests/models/test_evaluation.py [hunk 2]
```diff
@@ -644,12 +731,7 @@ def _load_content_from_file(self, local_artifact_path):
         raise RuntimeError()
 
 
-class PyFuncModelMatcher:
-    def __eq__(self, other):
-        return isinstance(other, mlflow.pyfunc.PyFuncModel)
```
Comment (harupy) [inline tests/models/test_evaluation.py:649]: Let's revert this :)

File: tests/models/test_evaluation.py [hunk 3]
```diff
@@ -710,10 +793,60 @@ def test_evaluator_interface(multiclass_logistic_regressor_model_uri, iris_datas
...
+    multiclass_logistic_regressor_model_uri, iris_dataset, baseline_model_uri, expected_error
+):
+    with mock.patch.object(
+        _model_evaluation_registry, "_registry", {"test_evaluator1": FakeEvauator1}
+    ):
+        evaluator1_config = {"config": True}
+        classifier_model = mlflow.pyfunc.load_model(multiclass_logistic_regressor_model_uri)
+        with expected_error:
+            evaluate(
+                classifier_model,
```
Comment (harupy) [inline tests/models/test_evaluation.py:830]: I think this line needs to be fixed. `evaluate` no longer accepts a pyfunc model.

**Review by harupy — APPROVED** [review]
LGTM once remaining comments are addressed!

**Comment by harupy** [comment]
We can remove files in `tests/pylint_plugins` :)

**Comment by dbczumar** [comment]
> Should we revert the changes in examples/pipelines/sklearn_regression?

No, we should update the pipelines submodule to the same commit used on `master`, which this PR does.

**Comment by dbczumar** [comment]
> tests/pylint_plugins

Thanks! Removed!

File: tests/models/test_evaluation.py (hunk 2 above)
Comment (dbczumar) [reply tests/models/test_evaluation.py:649]: Thanks for catching this - reverted!

File: tests/models/test_evaluation.py (hunk 3 above)
Comment (dbczumar) [reply tests/models/test_evaluation.py]: Fixed!

**Pull request merged**